hw2_function/fastq_filtrator.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[51]:
+
+
+#Counting GC content
+def GC_comp(seq):
+    GC = ((seq.upper().count('C') + seq.upper().count('G'))/len(seq.strip()))*100
+    GC = round(GC, 2)
+    return (GC)
+
+#Counting average quality
+def quality(seq):
+    av = 0
+    for sym in seq:
+        q_score = ord(sym)-33
+        av += q_score
+    av = round(av/len(seq.strip()), 2)
+    return(av)
+
+#Check for file existence 
+def is_accessible(path, mode='r'):
+    try:
+        f = open(path, mode)
+        f.close()
+    except IOError:
+         return False
+    return True
+
+def main (input_fastq, output_file_prefix, gc_bounds = (0, 100), length_bounds = (0, 2**32),
+          quality_threshold = 0, save_filtered = False):
+    
+    if type(gc_bounds) != tuple:
+        gc_bounds = (0, gc_bounds)
+    if type(length_bounds) != tuple:
+        length_bounds = (0, length_bounds)
+        
+    output_passed = output_file_prefix + '_passed.fastq'
+    if is_accessible(output_passed, mode='r') == True:
+        f = open(output_passed, 'w')
+        f.close
```

Comment:
1. `is_accessible` уже возвращает логический тип, сравнивать не обязательно
2. Ты не вызвала метод `close` (не написала скобочки), поэтому файл не закрываеься
```suggestion
    if is_accessible(output_passed, mode='r'):
        f = open(output_passed, 'w')
        f.close()
```